### PR TITLE
fix: do not fail parsing of .Net projects when dependency version is …

### PR DESCRIPTION
**Pull request by kyegupov**
…a variable not found in manifest.

- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/dotnet-deps-parser/blob/master/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?

We also collect dependencies with unknown versions into a new field of the `PkgTree` type

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/BST-407

### CODE REVIEW

**Comment by kyegupov** [comment]
@miiila @lili2311 do you know anything about Travis build failure on Node 8?

File: lib/parsers/index.ts [hunk 1]
```diff
@@ -4,14 +4,20 @@ import {InvalidUserInputError} from '../errors';
...
-  version: string;
+  version: string | null;
   dependencies: {
     [dep: string]: PkgTree;
   };
   depType?: DepType;
   hasDevDependencies?: boolean;
   cyclic?: boolean;
   targetFrameworks?: string[];
+  dependenciesWithUnknownVersions?: string[];
```
[the remark this answers is not in the document]
Comment (kyegupov) [reply lib/parsers/index.ts]: As far as I know, quiet ignore for now.

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "@types/xml2js": "0.4.3",
     "lodash": "^4.17.11",
     "source-map-support": "^0.5.7",
+    "tsc": "^1.20150623.0",
```
[the remark this answers is not in the document]
Comment (kyegupov) [reply package.json]: Good catch; this is a botched typescript installation.

File: lib/parsers/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lili2311) [reply lib/parsers/index.ts]: https://snyksec.atlassian.net/browse/BST-500

**Pull request merged**